projects/packages/woocommerce-analytics/src/API/class-wc-analytics-tracking-proxy.php

```diff
@@ -55,6 +55,18 @@ public function register_routes() {
 	 * @return \WP_REST_Response|\WP_Error Response object on success, or WP_Error object on failure.
 	 */
 	public function track_events( $request ) {
+		// Check consent before processing any events
+		if ( ! Consent_Manager::has_analytics_consent() ) {
+			return new \WP_REST_Response(
+				array(
+					'success' => true,
```

Comment:
I'm on a fence on whether we should return true or false for the request, but I think true is fine as it shouldn't really happen.